networking_aci/plugins/ml2/drivers/mech_aci/allocations_manager.py

```diff
@@ -305,3 +339,20 @@ def _process_host_or_hostgroup(self, session, host, host_config, allocations, le
             alloc = AllocationsModel(host=host, level=level, segment_type=segment_type, segmentation_id=segmentation_id,
                                      network_id=None)
             session.add(alloc)
+
+    def _sync_hostgroup_modes(self):
+        LOG.info("Preparing hostgroup modes sync")
+
+        session = db_api.get_writer_session()
+        with session.begin(subtransactions=True):
+            # fetch all mode-hostgroups from db
+            db_groups = []
+            for db_entry in (session.query(HostgroupModeModel).with_lockmode('update')):
```

Comment:
with_lockmode seems to be depracted since sqlalchemy 0.9: https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.with_lockmode